### PR TITLE
Fix and improve TACA for handling QC runs

### PR DESCRIPTION
**Pull request by kedhammar**
- Make sure TACA can handle runs that generate NO sequencing data at all
- Refactor logic of control function to reduce complexity
  - Introduce custom Exception for quiet skipping (waiting on) runs 
- Improve documentation
- Minor polishing of test to pass

Tested in isolated environment on `ngi-preproc` for handling a bunch of QC runs.

### CODE REVIEW

**Comment by kedhammar** [comment]
I'll revisit this to fix failing lints and coverage next week after deploying #418 and addressing any mergre conflicts.

File: taca/analysis/analysis_nanopore.py [hunk 1]
```diff
@@ -153,19 +151,29 @@ def process_qc_run(ont_qc_run: ONT_qc_run):
...
-        logger.info(f"{ont_qc_run.run_name}: Asserting run contents...")
-        ont_qc_run.assert_contents()
+        raise WaitForRun(f"{ont_qc_run.run_name}: Run is not fully synced, skipping.")
 
-        # Update StatusDB
-        logger.info(f"{ont_qc_run.run_name}: Updating StatusDB...")
-        ont_qc_run.update_db_entry()
+    # Assert all files are in place
+    logger.info(f"{ont_qc_run.run_name}: Asserting run contents...")
+    ont_qc_run.assert_contents()
```
Comment (alneberg) [inline taca/analysis/analysis_nanopore.py:158]: Missing files here would raise assertionErrors. I guess it could be presented in a nicer way to the user?

**Review by alneberg — APPROVED** [review]
Looks good!

File: taca/analysis/analysis_nanopore.py (hunk 1 above)
Comment (alneberg) [reply taca/analysis/analysis_nanopore.py:158]: Forget about it, you catch the exception nicely outside of the method.

**Review by alneberg — APPROVED** [review]
Looks good! 👍

**Pull request merged**